clang/test/Driver/bounds-safety-unused.c

```diff
@@ -0,0 +1,12 @@
+// This reports a warning to follow the default behavior of ClangAs.
+// RUN: %clang -fexperimental-bounds-safety -x assembler -c %s -o /dev/null 2>&1 | FileCheck -check-prefix WARN %s
```

Comment:
@rapidsna Maybe there should be a version of this part of the test that doesn't pass `-x <lang>` and instead passes to a `.s` file to make sure the right thing happens when in that case?

Reply:
I think `-x assembler` and `-x assembler-with-cpp` are more specific and should cover what I want. 

`.s` must be `-x assembler` (except for Darwin which does `-x assembler-with-cpp` for `.s`), but that specific behavior should be tested separately but not for this PR.